tests/Unit/Router/RouterRedirectTest.php

```diff
@@ -110,7 +110,13 @@ static function (Routes $routes) use ($destination, $statusCode, $method): void
             },
         );
 
-        self::assertNull($testHeaders);
+        self::assertSame([
```

Comment:
For the next ErrorHandling feature I would move this test to a new class. There are also other tests such this one: https://github.com/gacela-project/router/blob/80dd2fa300fdcf01a1749aef3cb2d40baab5891b/tests/Unit/Router/RouterMatchTest.php#L29 that with this implementation is now not testing the desired behavior.